src/utils/tutorial/index.js

```diff
@@ -0,0 +1,30 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import StringConstant from '../string/StringConstant';
+import {TUTOR_CREATE_POST_ANON, TUTOR_POST_ANON} from '../constants';
+
+// eslint-disable-next-line consistent-return
+export const getStorageKey = async (name) => {
+  switch (name) {
+    case StringConstant.tutorialAnonymousPostTitle:
+      return TUTOR_POST_ANON;
+    case StringConstant.tutorialCreateAnonymousPostTitle:
+      return TUTOR_CREATE_POST_ANON;
+    default:
+  }
+};
+
+export const setFinishedTutor = async (key) => {
+  // await AsyncStorage.setItem(key, true);
+};
+
+export const getFinishedTutor = async (key) => {
+  const res = await AsyncStorage.getItem(key);
+  return Boolean(res);
+};
+
+export const handleTutorialFinished = async (name) => {
+  const key = await getStorageKey(name);
+  if (key) {
+    setFinishedTutor(key);
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `handleTutorialFinished` function, you're calling `setFinishedTutor(key)` without awaiting it. Since `setFinishedTutor` is an async function, it should be awaited to ensure it completes before moving on.

```diff
-   setFinishedTutor(key);
+   await setFinishedTutor(key);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->